### ISSUE TITLE
ActiveRecord form with two submit buttons doesn't post name of submit button

### ISSUE THREAD

**Issue by smohadjer** [yiisoft/yii2#17808]
I have an "ActiveRecord" model named "Events" and a view with two submit buttons like below (generated HTML code in browser):

```
<form id="starterWizard__form" class="starterWizard__form" action="/wizard/step1?id=19623" method="post"></form>
<!-- my form fields -->
<input type="submit" class="btn btn-primary" name="Events[submitbutton]" value="Save and Exit" form="starterWizard__form">
<input type="submit" class="btn btn-primary" name="Events[submitbutton]" value="Continue" form="starterWizard__form">
```
In my controller I want to implement different logic based on which submit button is used to post the form. But the problem is that I don’t see “submitbutton” in $_POST.  If I test it in a plain php file without Yii then it works. I have also noticed the issue seems to be due to javascript that Yii runs in the page. If I comment out the following line of script in the page, then problem gets resolved.

`jQuery('#starterWizard__form').yiiActiveForm([], []);
`

I have tried generating markup for submit buttons using both ActiveField and Yii Helpers and neither works. Below is my Yii code for submit buttons:

```
$form->field($model, 'submitbutton', [
		'template' => '{input}',
		'inputOptions' => [
			'form' => 'starterWizard__form',
			'type' => 'submit',
			'value' => 'Save',
			'class' => 'btn btn-primary'
		]
	]);

Html::submitInput('Continue >>', [
	'class' => 'btn btn-primary',
	'form' => 'starterWizard__form',
	'name' => 'Events[submitbutton]',
	'value' => 'continue'
]);
```

**Comment by alex-code** [comment]
This PR should fix this, https://github.com/yiisoft/yii2/pull/17806
It's to do with the submit button being outside the form.

**Comment by smohadjer** [comment]
I don’t think that pull is related. My form is being submitted when I click the submit button, only the name of submit button is not being posted. Why is even javascript involved in form submission? I have disabled client-side validation and do not need any javascript functionality. 

Would it work if I do not use ActiveForm widget and build my view using Yii’s HTML helpers or pure html? Then the ActiveForm script will no longer be loaded on my page, will it? My ActiveRecord model can remain as before as long as my view refers to my model attributes with proper names in my markup.

**Comment by alex-code** [comment]
It's the way `ActiveForm` works, It intercepts submits then tries to recreate the button later but doesn't because it wasn't inside the form.

I tested the PR and it worked for me, button values were submitted.

If you don't need validation you could just use the helpers like you said.

**Comment by smohadjer** [comment]
But how come the form is submitted when there is only one submit button despite the button sitting outside the form?!

**Comment by alex-code** [comment]
The button (even if outside the form) triggers the submit.
The submit is caught then validation is tried.
After validating (using promises) the submit needs to be triggered again to send the POST. If the button that triggered the submit has `name` and `value` attributes these are added as a hidden element to be included as part of the POST.

The issue is if the button is outside the form it can trigger the submit but it's attributes aren't carried over.

**Comment by smohadjer** [comment]
Yes, the pr fixes this issue, but does it mean that ActiveForm doesn't work without JavaScript? My app must work even when JavaScript is disabled, so I guess I need to stop using ActiveRecord, right? :(

**Comment by alex-code** [comment]
If JS is disabled the form should just submit like normal, it just won't do any clientside validation.

**Comment by smohadjer** [comment]
If all the script does is clientside validation, why is it at all running considering that I have set enableClientValidation to false in my ActiveForm options? Shouldn’t the script only be loaded if clientside validation is desired?

**Comment by alex-code** [comment]
If you don't want any `ActiveForm` JS loaded then use this property, https://www.yiiframework.com/doc/api/2.0/yii-widgets-activeform#$enableClientScript-detail

**Comment by smohadjer** [comment]
Thanks, I will try that. Any idea when the pull request will get released?

**Comment by samdark** [comment]
@smohadjer when tests verifying behavior will be added to it.
